pkg/endpoints/pipeline_test.go

```diff
@@ -14,1408 +14,1039 @@ limitations under the License.
+	for i, _ := range taskRuns {
+		key := fmt.Sprintf("TaskRunStatus%d", i)
+		// The tasks within this pipelineRunTaskRunStatus do no exist
+		pipelineRunTaskRunStatus[key] = &v1alpha1.PipelineRunTaskRunStatus{
+			PipelineTaskName: fmt.Sprintf("Task%d", i),
+			Status:           &taskRuns[i].Status,
+		}
 	}
-	Pod2 := corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "Pod2",
-		},
-		Spec: corev1.PodSpec{
-			Containers:     []corev1.Container{Container5},
-			InitContainers: []corev1.Container{Container6},
+
+	pipelineName := "PipelineRun1"
+	pipelineRuns := []v1alpha1.PipelineRun{
+		v1alpha1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: pipelineName,
+			},
+			Spec: v1alpha1.PipelineRunSpec{},
+			Status: v1alpha1.PipelineRunStatus{
+				TaskRuns: pipelineRunTaskRunStatus,
+			},
 		},
 	}
-	Pod3 := corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "Pod3",
-		},
-		Spec: corev1.PodSpec{
-			Containers:     []corev1.Container{Container7},
-			InitContainers: []corev1.Container{},
-		},
+	for _, pipelineRun := range pipelineRuns {
+		_, err := r.PipelineClient.TektonV1alpha1().PipelineRuns(namespace).Create(&pipelineRun)
+		if err != nil {
+			t.Errorf("Error creating pipelineRun '%s': %v\n", pipelineRun.Name, err)
+		}
 	}
-	Pod4 := corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "Pod4",
-		},
-		Spec: corev1.PodSpec{
-			Containers:     []corev1.Container{},
-			InitContainers: []corev1.Container{},
-		},
+	// Test updatePipelineRun function: good request
+	httpRequestBody := strings.NewReader(`{"status" : "PipelineRunCancelled"}`)
+	httpReq := dummyHTTPRequest("PUT", fmt.Sprintf("%s/v1/namespaces/%s/pipelineruns/%s", server.URL, namespace, pipelineName), httpRequestBody)
+	response, err := http.DefaultClient.Do(httpReq)
+	if err != nil {
+		t.Errorf("Error getting updatePipelineRun response for pipelineRun '%s': %v\n", pipelineName, err)
+	} else {
+		expectedStatusCode := 204
+		if response.StatusCode != expectedStatusCode {
+			t.Errorf("Response code %d did not equal expected %d\n", response.StatusCode, expectedStatusCode)
+		} else { // Test duplicate cancel/update if successful
+			// Test updatePipelineRun function: second good request
+			httpRequestBody := strings.NewReader(`{"status" : "PipelineRunCancelled"}`)
+			httpReq := dummyHTTPRequest("PUT", fmt.Sprintf("%s/v1/namespaces/%s/pipelineruns/%s", server.URL, namespace, pipelineName), httpRequestBody)
+			response, err := http.DefaultClient.Do(httpReq)
+			if err != nil {
+				t.Errorf("Error getting updatePipelineRun response for pipelineRun '%s': %v\n", pipelineName, err)
+			} else {
+				expectedStatusCode := 412
+				if response.StatusCode != expectedStatusCode {
+					t.Errorf("Response code %d did not equal expected %d\n", response.StatusCode, expectedStatusCode)
+				}
+			}
+		}
 	}
-
-	// Add sample TaskRuns
-	_, err := r.PipelineClient.TektonV1alpha1().TaskRuns("ns1").Create(&TaskRun1)
+	// Test updatePipelineRun function: bad request
```

Comment:
This method currently does several different tests. Can they be split up into separate test methods?

Reply:
Addressed by below two responses